src/xsl/openrosa2html5form.xsl

```diff
@@ -28,20 +28,62 @@ XSLT Stylesheet that transforms OpenRosa style (X)Forms into valid HTMl5 forms
     <xsl:param name="openclinica"/>
     <xsl:output method="html" omit-xml-declaration="yes" encoding="UTF-8" indent="yes"/><!-- for xml: version="1.0" -->
 
+    <xsl:key
+        name="bindings-by-id"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@id" />
+    <xsl:key
+        name="nodeset-bindings"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@nodeset" />
+    <xsl:key
+        name="preload-bindings"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@jr:preload != ''" />
+    <xsl:key
+        name="calculate-bindings"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@calculate != ''" />
+    <xsl:key name="instances" match="/h:html/h:head/xf:model/xf:instance" use="true()" />
+    <xsl:key
+        name="primary-instance-root"
+        match="/h:html/h:head/xf:model/xf:instance[1]/child::node()"
+        use="true()" />
+    <xsl:key
+        name="model-actions"
+        match="/h:html/h:head/xf:model/xf:setvalue[@event] | /h:html/h:head/xf:model/odk:setgeopoint[@event]"
+        use="local-name()" />
+    <xsl:key
+        name="fields-by-ref"
+        match="/h:html/h:body//xf:input | /h:html/h:body//xf:upload | /h:html/h:body//xf:select | /h:html/h:body//xf:select1"
```

Comment:
This matches existing usage. But probably it intends to match all form control elements? https://getodk.github.io/xforms-spec/#body-elements